### DISCUSSION TITLE
What are the difference between deno native server and std/http

### DISCUSSION THREAD

**Discussion by Srabutdotcom** [oakserver/oak#350] [Q&A] [answered]
Options are confusing for anyone doesn't know like me.

What are the difference between deno native server and std/http?
Why the former is unstable?
And is the second truly stable for http server?
Which one is better?

Thanks

**Comment by kitsonk** [comment, marked as answer]
The difference is that `std/http` does all the processing of the HTTP request and response in JavaScript reading and writing directly to the network connection. While native HTTP processes the requests and responses in Rust using the hyper Rust crate.

The native HTTP is unstable because it is new, we are not 100% convinces that we won't need to introduce breaking changes to it. It also currently lacks the ability to upgrade HTTP connections to Web sockets, which is a fairly important feature that `std/http` supports.

Having said that `std` as a whole is unstable, and once native HTTP is in a place where it can be stabilised, `std/http` will be deprecated.

**Reply by Srabutdotcom** [reply]
thanks @kitsonk it is more clearer now for the user like me to choose between the two.
however i wonder why should deno need to create native http server in Rust as the second alternative.?
